tests/unittests/test_implicit.py

```diff
@@ -61,20 +65,21 @@ def dftb_calculator(device, skf_file: str):
     overlap_feed = SkFeed.from_database(skf_file, species, 'overlap', device=device)
     occupation_feed = SkfOccupationFeed.from_database(skf_file, species, device=device)
     hubbard_feed = HubbardFeed.from_database(skf_file, species, device=device)
-    repulsive_feed = RepulsiveSplineFeed.from_database(skf_file, species, device=device)
+    repulsive_feed = PairwiseRepulsiveEnergyFeed.from_database(skf_file, species, device=device)
 
     # set up the calculator
     return Dftb2(hamiltonian_feed, overlap_feed, occupation_feed,
                  hubbard_feed, r_feed=repulsive_feed, filling_scheme=None)
 
 
 def implicit_gradient_helper(mol: Geometry, orbs: OrbitalInfo, dftb_calculator, device):
-
+    dftb_calculator.grad_mode = "direct"
     energy_direct = dftb_calculator(mol, orbs, grad_mode='direct')
     forces_direct = -torch.autograd.grad(
         energy_direct, mol.positions,
         grad_outputs=torch.ones_like(energy_direct))[0]
 
+    dftb_calculator.grad_mode = "implicit"
     energy_imp = dftb_calculator(mol, orbs, grad_mode='implicit')
```

Comment:
energy_imp = dftb_calculator(mol, orbs, ~~grad_mode='implicit'~~)

Reply:
Hey Adam, since we have changed `grad_mode` as an argument of the calculator, we also need to change corresponding definitions of calculators in our examples.
For example: 
https://github.com/mcsloy/tbmalt_new/blob/29b657958c6e721e9af9c683339d2c529f346821/examples/example_01/example_01.py#L98
https://github.com/mcsloy/tbmalt_new/blob/29b657958c6e721e9af9c683339d2c529f346821/examples/example_01/example_01.py#L176
I can make a new pull request later to fix it :)

Reply:
Ah yes, you are right, there are a few of those that I have missed in various places. I will remove the `**kwargs` option from the `forward` method so that unknown arguments raise an error, and I will locate and remove all instance where `grad_mode` is specified in the calculator run call.